tests/workloads/test_dataframe.py

```diff
@@ -0,0 +1,60 @@
+from dask.sizeof import sizeof
+from dask.utils import format_bytes
+
+from ..utils_test import cluster_memory, timeseries_of_size, wait
+
+
+def print_dataframe_info(df):
+    p = df.partitions[0].compute(scheduler="threads")
+    partition_size = sizeof(p)
+    total_size = partition_size * df.npartitions
+    print(
+        f"~{len(p) * df.npartitions:,} rows x {len(df.columns)} columns, "
+        f"{format_bytes(total_size)} total, "
+        f"{df.npartitions:,} {format_bytes(partition_size)} partitions"
+    )
+
+
+def test_dataframe_align(small_client):
```

Comment:
If I understand your intention here, these tests are aiming at growing to incorporate weak scaling. I'm thinking through how they will show up on the static site as it exists today: the current strategy is to take parameterized tests and show them on a single chart as multiple lines. I think that will work well for these, but I'm wondering if it would be clearer to the reader if they were named something like `test_dataframe_align_weak_scaling` etc. I suppose it's a bit of letting charts wag the dog, but doesn't offend me too much.

Reply:
> the current strategy is to take parameterized tests and show them on a single chart as multiple lines

Ah, I didn't realize that's how they'd show up. I was thinking we'd eventually want to test both strong and weak scaling, so annoyingly, I imagine we'd want a separate chart for the strong vs weak case (but still probably multiple lines for each cluster size within them?). I wonder if we can make some sort of special `scaling_type` fixture that `pytest.mark`s the different cases, and we could use those marks in the dashboard script to separate them?

Reply:
From an implementation standpoint, I would love to not have to rely on pytest magic (or even importing pytest) to generate any data visualizations (just the data in sqlite). If you wanted to test both strong and weak scaling *within the same test*, I might prefer to just parameterize with a magic `id` string that we can identify in the test name. (i.e., if we see `weak-scaling` and `strong-scaling` in the parameterized test name, make two charts).

Reply:
I wasn't saying that the dashboard script would use pytest magic. Just that we could use such pytest magic to parameterize the tests over strong vs weak scaling mode, then pull out that parameter and write it into the database, vaguely like:

```diff
diff --git a/conftest.py b/conftest.py
index 78fb427..6667f25 100644
--- a/conftest.py
+++ b/conftest.py
@@ -184,6 +184,7 @@ def test_run_benchmark(benchmark_db_session, request, testrun_uid):
             python_version=".".join(map(str, sys.version_info)),
             platform=sys.platform,
             ci_run_url=WORKFLOW_URL,
+            scaling=request.keywords.get("scaling", None),
         )
         yield run
 ```
Then the dashboard script could just group by the `scaling` column to render different scaling modes in different charts.

Reply:
This sounds like a good approach